### ISSUE TITLE
Failed to set OOM Score on shim error when running docker:dind in Kubernetes

### ISSUE THREAD

**Issue by skaegi** [moby/moby#42041]
**Description**
We're running `docker:dind` in a Kubernetes pod for CI. With the Docker 20 version of dind (which now uses the v2 shim) this is now broken and errors with a message similar to...
```
docker: Error response from daemon: io.containerd.runc.v2: failed to adjust OOM score for shim: set shim OOM score: write /proc/211/oom_score_adj: invalid argument
```
Also see -- https://github.com/containerd/containerd/issues/4837
Fixed in this PR -- https://github.com/containerd/containerd/pull/4845

Current containerd version is... https://github.com/moby/moby/blob/master/vendor.conf#L133

Could we bump to a more recent commit -- potentially https://github.com/containerd/containerd/releases/tag/v1.5.0-beta.1

Also, could this be backported for a Docker 20 minor release as we've seen many teams encounter issues and the current workaround is to go back to using Docker 19.

**Comment by thaJeztah** [comment]
https://github.com/moby/moby/pull/41528 removed the default oom-score adjust when running the daemon without systemd (which is the case when running the `docker:20.10-dind` image. How is the docker-in-docker container started? Is the container started with `--oom-score-adj` set? Does it work if you start the docker-in-docker container with a more reasonable score? (e.g. `--oom-score-adj -500` to use the previous default)? (also see https://github.com/docker-library/docs/pull/1821)

> Could we bump to a more recent commit -- potentially https://github.com/containerd/containerd/releases/tag/v1.5.0-beta.1

We'd have to evaluate changes in between, the diff between what's currently used and that commit is quite large;
https://github.com/containerd/containerd/compare/0edc412565dcc6e3d6125ff9e4b009ad4b89c638...v1.5.0-beta.1

Picking the first commit with that fix is somewhat smaller;

https://github.com/containerd/containerd/compare/0edc412565dcc6e3d6125ff9e4b009ad4b89c638...070b6984491715d701d53ce18c00d5dcb8d696e5

But both would have to be looked at to see what diff is introduced in the code we consume (also taking into account possible other dependency updates related to containerd)

**Comment by skaegi** [comment]
Sorry @thaJeztah just saw this now. Yes ... setting `--oom-score-adj -500` for `dockerd` works, as does setting a pod memory resource limit. Still would be good to get this fixed as many users (CI/CD) are not going to know/understand why.

I also totally understand what you're saying about the difficulty in evaluating a new containerd version but at least wanted to log the issue/request to track it.

**Comment by skaegi** [comment]
Now that ... https://github.com/containerd/containerd/releases/tag/v1.5.0 ... is out. That would definitely be the preferred version to sync with for us and likely many teams that are in environments where compliance is an issue.

**Comment by skaegi** [comment]
Fixed in master now -- but problem still present on 20.10 branch

**Comment by skaegi** [comment]
Actually this was fixed in the 20.10 branch back in March and released in April with 20.10.6 with https://github.com/moby/moby/pull/42124
